[VL] Static build issue when GCS is enabled

### Problem description

Here is error log in building abal (vcpkg installed version is not the required one), required by google cloud cpp. Looks raw_logging.h (an old version of absl) installed in vcpkg is included in compiling.
```
2024-06-04T03:02:17.6948676Z [6/982] Building CXX object _deps/absl-build/absl/base/CMakeFiles/raw_logging_internal.dir/internal/raw_logging.cc.o
2024-06-04T03:02:17.6951327Z FAILED: _deps/absl-build/absl/base/CMakeFiles/raw_logging_internal.dir/internal/raw_logging.cc.o 
2024-06-04T03:02:17.6961341Z /opt/rh/devtoolset-9/root/usr/bin/c++ -DFOLLY_HAVE_INT128_T=1 -DVELOX_ENABLE_ABFS -DVELOX_ENABLE_HDFS3 -DVELOX_ENABLE_PARQUET -DVELOX_ENABLE_S3 -I/__w/incubator-gluten/incubator-gluten/dev/vcpkg/vcpkg_installed/x64-linux-avx/include -isystem /__w/incubator-gluten/incubator-gluten/ep/build-velox/build/velox_ep/_build/release/_deps/absl-src -Wno-missing-field-initializers -march=native -std=c++17 -mno-avx512f -mbmi2 -D USE_VELOX_COMMON_BASE -D HAS_UNCAUGHT_EXCEPTIONS -Wall -Wextra -Wno-unused        -Wno-unused-parameter        -Wno-sign-compare        -Wno-ignored-qualifiers        -Wno-implicit-fallthrough          -Wno-class-memaccess          -Wno-comment          -Wno-int-in-bool-context          -Wno-redundant-move          -Wno-array-bounds          -Wno-maybe-uninitialized          -Wno-unused-result          -Wno-format-overflow          -Wno-strict-aliasing -O3 -DNDEBUG -std=gnu++17 -fPIC -fdiagnostics-color=always -Wall -Wextra -Wcast-qual -Wconversion-null -Wformat-security -Wmissing-declarations -Woverlength-strings -Wpointer-arith -Wundef -Wunused-local-typedefs -Wunused-result -Wvarargs -Wvla -Wwrite-strings -DNOMINMAX -MD -MT _deps/absl-build/absl/base/CMakeFiles/raw_logging_internal.dir/internal/raw_logging.cc.o -MF _deps/absl-build/absl/base/CMakeFiles/raw_logging_internal.dir/internal/raw_logging.cc.o.d -o _deps/absl-build/absl/base/CMakeFiles/raw_logging_internal.dir/internal/raw_logging.cc.o -c /__w/incubator-gluten/incubator-gluten/ep/build-velox/build/velox_ep/_build/release/_deps/absl-src/absl/base/internal/raw_logging.cc
2024-06-04T03:02:17.6977549Z /__w/incubator-gluten/incubator-gluten/ep/build-velox/build/velox_ep/_build/release/_deps/absl-src/absl/base/internal/raw_logging.cc: In function 'void absl::lts_20230125::raw_log_internal::{anonymous}::RawLogVA(absl::lts_20230125::LogSeverity, const char*, int, const char*, __va_list_tag*)':
2024-06-04T03:02:17.6981062Z /__w/incubator-gluten/incubator-gluten/ep/build-velox/build/velox_ep/_build/release/_deps/absl-src/absl/base/internal/raw_logging.cc:178:5: error: 'AsyncSignalSafeWriteError' was not declared in this scope; did you mean 'AsyncSignalSafeWriteToStderr'?
2024-06-04T03:02:17.6982922Z   178 |     AsyncSignalSafeWriteError(buffer, strlen(buffer));
2024-06-04T03:02:17.6983923Z       |     ^~~~~~~~~~~~~~~~~~~~~~~~~
2024-06-04T03:02:17.6984493Z       |     AsyncSignalSafeWriteToStderr
2024-06-04T03:02:17.6986524Z /__w/incubator-gluten/incubator-gluten/ep/build-velox/build/velox_ep/_build/release/_deps/absl-src/absl/base/internal/raw_logging.cc: At global scope:
2024-06-04T03:02:17.6992358Z /__w/incubator-gluten/incubator-gluten/ep/build-velox/build/velox_ep/_build/release/_deps/absl-src/absl/base/internal/raw_logging.cc:206:6: warning: no previous declaration for 'void absl::lts_20230125::raw_log_internal::AsyncSignalSafeWriteError(const char*, size_t)' [-Wmissing-declarations]
2024-06-04T03:02:17.6995703Z   206 | void AsyncSignalSafeWriteError(const char* s, size_t len) {
```

### System information

N/A

### CMake log

_No response_

--- a/dev/ci-velox-buildstatic.sh
+++ b/dev/ci-velox-buildstatic.sh
@@ -6,4 +6,4 @@ cd $GITHUB_WORKSPACE/
 source ./dev/vcpkg/env.sh
 sed -i '/^headers/d' ep/build-velox/build/velox_ep/CMakeLists.txt
 export NUM_THREADS=4
-./dev/builddeps-veloxbe.sh --build_tests=OFF --build_benchmarks=OFF --enable_s3=ON --enable_gcs=OFF --enable_hdfs=ON --enable_abfs=ON
+./dev/builddeps-veloxbe.sh --build_tests=OFF --build_benchmarks=OFF --enable_s3=ON --enable_gcs=ON --enable_hdfs=ON --enable_abfs=ON

--- a/dev/vcpkg/vcpkg-configuration.json
+++ b/dev/vcpkg/vcpkg-configuration.json
@@ -8,6 +8,13 @@
       "repository": "https://github.com/Microsoft/vcpkg",
       "baseline": "8ccb84df727bdf83045e53c319af05c554838b80",
       "packages": [ "boost*", "boost-*"]
+    },
+    {
+      "kind": "git",
+      "repository": "https://github.com/Microsoft/vcpkg",
+      "baseline": "f82160a747b1b529daad7ce2f75a0d6c6a0d285d",
+      "packages": [ "google-cloud-cpp"]
     }
+
   ]
 }

--- a/dev/vcpkg/vcpkg.json
+++ b/dev/vcpkg/vcpkg.json
@@ -76,7 +76,8 @@
           "name": "google-cloud-cpp",
           "features": [
             "storage"
-          ]
+          ],
+          "version>=": "2.22.0"
         }
       ]
     },

